### ISSUE TITLE
Checkout page form is not validated

### ISSUE THREAD

**Issue by wAsnk** [OrchardCMS/OrchardCore.Commerce#218]
### Describe the bug

The checkout page form is not validated, thus required fields can remain empty and still pay and create an order.

### To Reproduce

1. Add a product to cart proceed to the /checkout page.
2. Do not fill required fields. e.g. Phone
3. Try to pay
4. You can, without an error for missing phone

### Expected behavior

Validate the form before paying.

**Comment by wAsnk** [comment]
This bug might came from using a custom shape instead of the OrderPart edit shape. https://github.com/OrchardCMS/OrchardCore.Commerce/issues/211

**Comment by DAud-IcI** [comment]
You are right, the _~/success_ POST action doesn't check the model state after `IContentItemDisplayManager.UpdateEditorAsync()` is called, as it should. But I think the workflow was bad even before that: we don't want to fail the order creation after the payment has already went through. So we have to introduce an async form validation step before payment as well as fix the validation in _~/success_ to safeguard against (potentially malicious) changes between payment and success.
